syntax = "proto3";

package product;

service ProductService {
  rpc GetProduct (GetProductRequest) returns (ProductResponse);
  rpc UpdateProduct (UpdateProductRequest) returns (ProductResponse);
}

message Product {
  string id = 1;
  string name = 2;
  float price = 3;
  string description = 4;
}

message GetProductRequest {
  string id = 1;
  repeated string fields = 2;
}

message UpdateProductRequest {
  string id = 1;
  string name = 2;
  float price = 3;
  string description = 4;
  repeated string fields = 5;
}

message ProductResponse {
  Product product = 1;
}
